chia/wallet/wallet_weight_proof_handler.py

```diff
@@ -67,100 +55,30 @@ def cancel_weight_proof_tasks(self):
     async def validate_weight_proof(
         self, weight_proof: WeightProof, skip_segment_validation: bool = False, old_proof: Optional[WeightProof] = None
     ) -> Tuple[bool, List[SubEpochSummary], List[BlockRecord]]:
+        summaries, sub_epoch_weight_list = _validate_sub_epoch_summaries(self._constants, weight_proof)
+        await asyncio.sleep(0)  # break up otherwise multi-second sync code
+        if summaries is None or sub_epoch_weight_list is None:
+            log.error("weight proof failed sub epoch data validation")
+            return False, [], []
         validate_from = get_fork_ses_idx(old_proof, weight_proof)
         task: asyncio.Task = asyncio.create_task(
-            self._validate_weight_proof_inner(weight_proof, skip_segment_validation, validate_from)
+            validate_weight_proof_inner(
+                self._constants,
+                self._executor,
+                self._executor_shutdown_tempfile.name,
+                self._num_processes,
+                weight_proof,
+                summaries,
+                sub_epoch_weight_list,
+                skip_segment_validation,
+                validate_from,
+            )
         )
         self._weight_proof_tasks.append(task)
-        valid, summaries, block_records = await task
+        valid, block_records = await task
         self._weight_proof_tasks.remove(task)
         return valid, summaries, block_records
 
-    async def _validate_weight_proof_inner(
```

Comment:
so this was basically completely duplicated?

Reply:
yes. i didn't change any logic regarding the WP validation,  just removed redundant code duplication 

all the changes related to the weight_proof are in https://github.com/Chia-Network/chia-blockchain/pull/12310 
i needed that so i can organise the code better but i think its better to review that there (Mariano also reviewed it)